gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala

```diff
@@ -364,6 +364,18 @@ case class GremlinScala[End, Labels <: HList](traversal: GraphTraversal[_, End])
     GremlinScala[A, Labels](traversal.union(mappedTraversals: _*))
   }
 
+  def choose[A, B](
+    predicate: GremlinScala[End, HNil] ⇒ GremlinScala[A, _],
+    trueChoice: GremlinScala[End, HNil] ⇒ GremlinScala[B, _],
```

Comment:
should we call the choice parameters `onTrue` and `onFalse`? just thinking aloud here, happy to keep as is if you prefer